packages/iris-grid/src/AdvancedFilterCreatorFilterItem.tsx

```diff
@@ -114,36 +133,42 @@ class AdvancedFilterCreatorFilterItem extends PureComponent {
     return '';
   }
 
-  constructor(props) {
+  constructor(props: AdvancedFilterCreatorFilterItemProps) {
     super(props);
 
     this.handleDelete = this.handleDelete.bind(this);
     this.handleTypeChange = this.handleTypeChange.bind(this);
     this.handleValueChange = this.handleValueChange.bind(this);
     this.typeDropdown = null;
 
-    const { selectedType, value, filterTypes } = props;
+    const { value = '', filterTypes, selectedType = filterTypes[0] } = props;
 
     this.state = {
-      selectedType: selectedType || filterTypes[0],
+      selectedType,
       value,
     };
   }
 
-  componentDidMount() {
-    this.typeDropdown.focus();
+  componentDidMount(): void {
+    this.typeDropdown?.focus();
   }
 
-  componentDidUpdate(prevProps) {
-    const { selectedType, value } = this.props;
+  componentDidUpdate(prevProps: AdvancedFilterCreatorFilterItemProps): void {
+    const {
+      value = '',
+      filterTypes,
+      selectedType = filterTypes[0],
+    } = this.props;
     if (prevProps.selectedType !== selectedType || prevProps.value !== value) {
       this.setState({ selectedType, value });
     }
   }
 
-  handleTypeChange(event) {
+  typeDropdown: HTMLSelectElement | null;
```

Comment:
We can. It only applies to jsx/tsx files, but the rule is `react/sort-comp`. We could add `instance-variables` before `lifecycle` and it would enforce it. Probably should be its own ticket since it needs fixes and it doesn't seem that one has auto-fix.

Reply:
Created #375 for that rule